src/resolvers/mutation.ts

```diff
@@ -385,3 +386,37 @@ export async function updateTag(
     );
   }
 }
+
+/**
+ * Replaces all tags for a given saved item with the tags provided in the input
+ * @param root
+ * @param args
+ * @param context
+ */
+export async function replaceSavedItemTags(
+  root,
+  args: { input: SavedItemTagsInput[] },
+  context: IContext
+): Promise<SavedItem[]> {
+  const savedItems = await new TagDataService(
+    context,
+    new SavedItemDataService(context)
+  ).replaceSavedItemTags(args.input);
+
+  const savedItemTagsMap = args.input.reduce((savedItemTags, input) => {
+    return {
+      ...savedItemTags,
+      [input.savedItemId]: input.tags,
+    };
+  }, {});
+
+  for (const savedItem of savedItems) {
```

Comment:
also looking  at this case :
```
[
  {savedItemIdA, [{tagA, tagB}]},
  {savedItemIdA, [{tagA, tagB}]},
  {savedItemIdA, [{tagA, tagC}]},
]
```
for this case,  i guess its ok to take the last record and ignore all previous one coz its a `replace`